src/solvers/dg/3d/dg.jl

```diff
@@ -50,7 +51,7 @@ mutable struct Dg3D{Eqn<:AbstractEquation, NVARS, POLYDEG,
   save_analysis::Bool
   analysis_filename::String
 
-  shock_indicator_variable::Symbol
+  shock_indicator_variable::ShockIndicatorVariable
```

Comment:
Why does this get its own type? Is it a *measurable* performance optimization?

Reply:
This makes it more similar to `mortar_type::MortarType`. In the future, I would like to get rid of this `Val{some_symbol}` dispatch and instead pass in functions, similar to the initial conditions etc. to make Trixi.jl more like a library that can be adapted by users without modifying the source code of Trixi.jl itself. This can be seen as as preparation step for it.